File: Ninja Slayground 2.0/11. Two Sum/Solution.java
// Learn the code, And copy and paste is not allowed. You are required to type...

// Follow this account, for all future code updates on GitHub.

import java.util.HashSet;
import java.util.Set;

public class Solution {
    public static String read(int n, int[] book, int target) {
        Set<Integer> seen = new HashSet<>();
        
        for (int pages : book) {
            int complement = target - pages;
            if (seen.contains(complement)) {
                return "YES";
            }
            seen.add(pages);
        }
        
        return "NO";
    }
}

File: Ninja Slayground 2.0/11. Two Sum/Solution.py
#Learn the code, Copy and Paste is not allowed. You are required to type...

#Follow this account, for all future code updates on GitHub.

def read(n: int, book: [int], target: int) -> str:
    seen = set()
    
    for pages in book:
        complement = target - pages
        if complement in seen:
            return "YES"
        seen.add(pages)
    
    return "NO"
